arduino-ide-extension/src/browser/contributions/open-sketch-files.ts

```diff
@@ -135,39 +130,29 @@ export class OpenSketchFiles extends SketchContribution {
     const widget = this.editorManager.all.find(
       (widget) => widget.editor.uri.toString() === uri
     );
+    if (widget && !forceOpen) {
+      return widget;
+    }
+
     const disposables = new DisposableCollection();
-    if (!widget || forceOpen) {
-      const deferred = new Deferred<EditorWidget>();
+    const deferred = new Deferred<EditorWidget>();
+    if (!widget) {
+      // If the editor or is not defined, assume one on create event
```

Comment:
I don't understand this comment. @kittaakos could you please elaborate on this?

Reply:
Thanks for asking!

We should improve it. I am open to any proposals. Your recent recommendations for clarifying code comments were on the spot 👍 

When `ensureOpened` is called, the editor can be in two different primary states:
 - It's not opened. The editor widget and the backing monaco editor instance do not exist, and the editor widget is not yet registered to the `editorManager`. The `widget` is `undefined`.
 - It's opened
    - the editor widget is the top editor. The editor has the focus (the cursor blinks in it); it's the [`activeEditor`](https://github.com/eclipse-theia/theia/blob/7abca9d57c966d621f2acc26008cf1a9e454949b/packages/editor/src/browser/editor-manager.ts#L138-L144).
    - the editor widget is the top editor, but the focus is on a different widget, or the context menu is active in the editor; it's the [`currentEditor`](https://github.com/eclipse-theia/theia/blob/7abca9d57c966d621f2acc26008cf1a9e454949b/packages/editor/src/browser/editor-manager.ts#L162-L168).
    - the editor widget is not on top. The tab with the filename is visible.

When the `widget` is `undefined`, the framework will instantiate it when calling `editorManager.open`. After the editor widget creation, `editorManager.onCreated` gets an event. IDE2 must handle this case differently, listen to this event, and ensure that the editor is attached to the shell and visible on the screen. Then resolve the promise.

If the `widget` is already defined, it's opened; hence, there is no need to wait for any visibility state changes. This case does not need to wait for any events.

Please help to improve the comment.

Reply:
I have updated the comments. Please re-check them. Thank you!